workitem/simple_type.go

```diff
@@ -94,6 +95,15 @@ func (fieldType SimpleType) ConvertToModel(value interface{}) (interface{}, erro
 		default:
 			return nil, errors.Errorf("value %v should be %s, but is %s", value, "MarkupContent", valueType)
 		}
+	case KindCodebase:
+		if valueType.Kind() != reflect.Map {
+			return nil, errors.Errorf("value %v should be %s, but is %s", value, reflect.Map, valueType.Name())
+		}
+		validCodebase, err := codebase.NewCodebase(value.(map[string]interface{}))
+		if err != nil {
+			return nil, err
+		}
+		return validCodebase.ToMap(), nil
```

Comment:
@aslakknutsen : this is `convertToModel` method and input from REST shold be validated in case of all invalid keys.. Earlier it was `ValidateCodebase` that returns the error but then I moved it to `NewCodebase`.
You are right that this conversion is from map to map, but in between it guarantees that newly generated map is as per need.
In case POST data does not have `repository` key then it can not cast it to Codebase.